### PR TITLE
transpile with babel

### PR DESCRIPTION
**Pull request by robertkowalski**
Hello ES6 and React 14! :sunny: 

Additional changes:

 - optimize watch task: run stylecheck before to provide faster
   feedback

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -376,39 +383,26 @@ module.exports = function (grunt) {
...
+    var targetFolder = folder;
 
-    // if the JS file that just changed was outside of Fauxton, copy it over
-    if (isJS && !(/^app\/addons/.test(filepath))) {
-      config.copy.changedFiles.files = [{
-        src: filepath,
-        dest: filepath.replace(/.*\/addons/, 'app/addons')
-      }];
-      grunt.task.run(['copy:changedFiles']);
-    }
```
Comment (robertkowalski) [inline Gruntfile.js]: i am not sure about the intention of the code code and it does not seem to make sense with all required js and compiled js files in the new dev server directory any more - can you confirm that @benkeen ?

Comment (benkeen) [inline Gruntfile.js]: I'm not sure. The purpose of this code was for local dev to ensure any changed JS files in the dash get copied over immediately (i.e. as you change them). Won't this condition still apply with the new folder structure?

**Comment by benkeen** [comment]
Super exciting. :D Can't wait to get React 14 in there.

From what I can see, this look good. It does break the dash, though ['cannot read property jQuery of  undefined` / `cannot set property Backbone` of undefined].

**Comment by robertkowalski** [comment]
@benkeen i had to disable the ES6 transpiling as it seems to break with phantomjs... so far haven't found out the issue.

it looks like nothing breaks now, can you retest?

**Comment by benkeen** [comment]
Looks good from here. There's a wayward console.log() in `routes-mango.js` but it looks like that snuck in  in another ticket. 

+1

**Comment by garrensmith** [comment]
+1 looks great.

**Comment by robertkowalski** [comment]
merged as cfc664d

**Pull request closed without merging**